src/main.rs

```diff
@@ -78,13 +76,17 @@ pub struct OsInputOutput {
     pub network_frames: Box<dyn DataLinkReceiver>,
     pub get_open_sockets: fn() -> HashMap<Connection, String>,
     pub keyboard_events: Box<dyn Iterator<Item = Event> + Send>,
-    pub lookup_addr: Box<dyn Fn(&IpAddr) -> Option<String> + Send>,
+    pub resolver_factory: Box<dyn FnOnce(IpTable) -> Result<Box<dyn Lookup>, failure::Error>>,
     pub on_winch: Box<dyn Fn(Box<dyn Fn()>) + Send>,
     pub cleanup: Box<dyn Fn() + Send>,
     pub write_to_stdout: Box<dyn FnMut(String) + Send>,
 }
 
-pub fn start<'a, B>(terminal_backend: B, os_input: OsInputOutput, opts: Opt)
+pub fn start<'a, B>(
+    terminal_backend: B,
+    os_input: OsInputOutput,
+    opts: Opt,
+) -> Result<(), failure::Error>
```

Comment:
Since the creation of the `Resolver` can fail I had to add the return type to the `start` function. I could maybe do it in the `try_main` function with the other fallible calls?

Reply:
I would prefer that - I feel it makes it simpler if the start function isn't supposed to fail.